test_app/lib/screens/participant_screen/conference_controls.dart

```diff
@@ -138,25 +138,15 @@ class _ConferenceControlsState extends State<ConferenceControls> {
   }
 
   void leaveConference({required bool closeSession})  {
-    if(closeSession) {
-        var options = ConferenceLeaveOptions(true);
-        _dolbyioCommsSdkFlutterPlugin.conference
-            .leave(options: options)
-            .then((value) {
-              Navigator.of(context).popUntil(ModalRoute.withName("LoginScreen"));
-            })
-            .onError((error, stackTrace) { onError('Error during leaving', error); });
-      } else {
-        var options = ConferenceLeaveOptions(false);
-        widget.conference.then((current) {
-          _dolbyioCommsSdkFlutterPlugin.conference
-              .leave(options: options)
-              .then((value) {
-                Navigator.of(context).popUntil(ModalRoute.withName("JoinConferenceScreen"));
-              })
-              .onError((error, stackTrace) {onError('Error during leaving', error);});
-        });
-      }
+    if (closeSession) {
+      widget.updateCloseSessionFlag(true);
+      Navigator.of(context).popUntil(ModalRoute.withName("LoginScreen"));
+    } else {
+      widget.updateCloseSessionFlag(false);
+
+      Navigator.of(context)
+          .popUntil(ModalRoute.withName("JoinConferenceScreen"));
```

Comment:
Now this contains so little code that we can move it to the button handlers.

Reply:
I updated it.